curp/src/client.rs

```diff
@@ -635,55 +708,100 @@ where
         use_fast_path: bool,
     ) -> Result<(C::ER, Option<C::ASR>), CommandProposeError<C>> {
         let cmd_arc = Arc::new(cmd);
+        loop {
+            let res_option = if use_fast_path {
+                self.fast_path(Arc::clone(&cmd_arc)).await
+            } else {
+                self.slow_path(Arc::clone(&cmd_arc)).await
+            };
+            let Some(res) = res_option else {
+                let cluster = self.fetch_cluster(false).await?;
+                self.set_cluster(cluster).await?;
```

Comment:
I think the `fetch_cluster` here need to make sure that the new cluster version is applied on superquorum of servers, otherwise the fast path will probably still fail after that fetch.

Reply:
When the argument passed in fetch cluster is true, it will fetch the latest cluster from leader